src/app/lists/ListStates.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import Menu from "@mui/material/Menu/Menu";
+import MenuList from "@mui/material/MenuList/MenuList";
+import MenuItem from "@mui/material/MenuItem/MenuItem";
+
+export type ListName = "Regular" | "Hotel";
+interface Props {
+    listStateAnchorElement: HTMLElement | null;
+    setListStateAnchorElement: React.Dispatch<React.SetStateAction<HTMLElement | null>>;
+    currentList: ListName;
+    setCurrentList: React.Dispatch<React.SetStateAction<ListName>>;
+}
+
+const swapListNameValue = (currentListName: ListName): ListName => {
+    return currentListName === "Regular" ? "Hotel" : "Regular";
+};
+
+const ListStates: React.FC<Props> = ({
+    listStateAnchorElement,
+    setListStateAnchorElement,
+    currentList,
+    setCurrentList,
+}) => {
+    return (
+        <>
+            {listStateAnchorElement && (
+                <Menu
+                    open
+                    onClose={() => setListStateAnchorElement(null)}
+                    anchorEl={listStateAnchorElement}
+                >
+                    <MenuList id="listType-menu">
+                        <MenuItem
+                            key={swapListNameValue(currentList)}
+                            onClick={() => {
+                                setCurrentList(swapListNameValue(currentList));
+                                setListStateAnchorElement(null);
+                            }}
+                        >
+                            {swapListNameValue(currentList)}
+                        </MenuItem>
+                    </MenuList>
+                </Menu>
+            )}
+        </>
```

Comment:
If you have a conditional rendering at the top level based on an optional argument, it should live outside the component. That means that `ListStates` should have `listStateAnchorElement` as a mandatory property, there should be no conditional here, then it should be called as
```
{listStateAnchorElement &&
    <ListStates {...props} />
}
```